dlc-oracle-test/src/test/scala/org/bitcoins/dlc/oracle/DLCOracleTest.scala

```diff
@@ -0,0 +1,121 @@
+
+  it must "correctly initialize" in { dlcOracle: DLCOracle =>
+    assert(dlcOracle.conf.seedExists())
+  }
+
+  it must "start with no events" in { dlcOracle: DLCOracle =>
+    dlcOracle.listEventDbs().map { events =>
+      assert(events.isEmpty)
+    }
+  }
+
+  it must "start with no pending events" in { dlcOracle: DLCOracle =>
+    dlcOracle.listPendingEventDbs().map { events =>
+      assert(events.isEmpty)
+    }
+  }
+
+  it must "create a new event and list it with pending" in {
+    dlcOracle: DLCOracle =>
+      for {
+        testEventDb <- dlcOracle.createNewEvent("test", testOutcomes)
+        pendingEvents <- dlcOracle.listPendingEventDbs()
+      } yield {
+        assert(pendingEvents.size == 1)
+        assert(pendingEvents.contains(testEventDb))
+      }
+  }
+
+  it must "create a new event with a valid commitment signature" in {
+    dlcOracle: DLCOracle =>
+      for {
+        testEventDb <- dlcOracle.createNewEvent("test", testOutcomes)
+        rValDbOpt <- dlcOracle.rValueDAO.read(testEventDb.nonce)
+      } yield {
+        assert(rValDbOpt.isDefined)
+        val rValDb = rValDbOpt.get
+        val hash = CryptoUtil.sha256(
+          rValDb.nonce.bytes ++ CryptoUtil.serializeForHash(rValDb.label))
+        assert(
+          dlcOracle.publicKey.verify(hash.bytes, rValDb.commitmentSignature))
+      }
+  }
+
+  it must "create multiple events with different names" in {
+    dlcOracle: DLCOracle =>
+      for {
+        _ <- dlcOracle.createNewEvent("test", testOutcomes)
+        _ <- dlcOracle.createNewEvent("test1", testOutcomes)
+      } yield succeed
+  }
+
+  it must "fail to create multiple events with the same name" in {
+    dlcOracle: DLCOracle =>
+      recoverToSucceededIf[SQLException] {
+        for {
+          _ <- dlcOracle.createNewEvent("test", testOutcomes)
+          _ <- dlcOracle.createNewEvent("test", testOutcomes)
+        } yield ()
+      }
+  }
+
+  it must "create and sign a event" in { dlcOracle: DLCOracle =>
+    val outcome = testOutcomes.head
+    for {
+      eventDb <- dlcOracle.createNewEvent("test", testOutcomes)
+      sig <- dlcOracle.signEvent(eventDb.nonce, outcome)
+      outcomeDbs <- dlcOracle.eventOutcomeDAO.findByNonce(eventDb.nonce)
+      outcomeDb = outcomeDbs.find(_.message == outcome).get
+      signedEvent <- dlcOracle.eventDAO.read(eventDb.nonce)
+    } yield {
+      assert(signedEvent.isDefined)
+      assert(signedEvent.get.attestationOpt.contains(sig.sig))
+      assert(dlcOracle.publicKey.verify(outcomeDb.hashedMessage.bytes, sig))
+      assert(
+        SchnorrDigitalSignature(signedEvent.get.nonce,
+                                signedEvent.get.attestationOpt.get) == sig)
+    }
+  }
+
+  it must "fail to sign an event that doesn't exist" in {
+    dlcOracle: DLCOracle =>
+      val dummyNonce = SchnorrNonce(ECPublicKey.freshPublicKey.bytes.tail)
+      recoverToSucceededIf[RuntimeException](
+        dlcOracle.signEvent(dummyNonce, "testOutcomes"))
+  }
```

Comment:
So I would like to see a test case that calls `createEvent()` twice in parallel a verify that the same `nonce` is NEVER returned. In general, we should always make sure the database configuration for the DLC oracle project has `numThreads=1` to make sure that there is no concurrent database accesses. We should use a `require()` inside of `DLCOracleAppConfig` to enforce this.

Reply:
Issue filed for the latter half of this issue: https://github.com/bitcoin-s/bitcoin-s/issues/2107